pkg/clustermesh/remote_cluster.go

```diff
@@ -177,6 +184,21 @@ func (rc *remoteCluster) restartRemoteConnection(allocator RemoteIdentityWatcher
 
 				rc.getLogger().Info("Connection to remote cluster established")
 
+				config, err := GetClusterConfig(rc.name, backend)
+				if err != nil && config == nil {
+					rc.getLogger().Warning("Remote cluster doesn't have cluster configuration, falling back to the old behavior")
```

Comment:
Let's add one more sentence explaining that to the user that this is expected when performing an upgrade where the old version doesn't have this new behavior implemented.

Reply:
Yep, good idea. I added.